File: prs-ai-hub/shared/prs_models/prs_models/sku.py
from pydantic import BaseModel, field_validator


class SKUItem(BaseModel):
    sku_number: str
    description: str
    unit_of_measure: str
    unit_price: float
    msrp: float | None = None
    min_order_qty: int
    lead_time: str
    status: str

    @field_validator("unit_price", "msrp")
    @classmethod
    def must_be_positive(cls, v: float | None) -> float | None:
        if v is not None and v <= 0:
            raise ValueError("Must be a positive number")
        return v


class SKUAddendumRequest(BaseModel):
    request_id: str
    sku_items: list[SKUItem]
    addendum_text: str

File: prs-ai-hub/frontend/lib/sku.ts
export type SKUItem = {
  sku_number: string;
  description: string;
  unit_of_measure: string;
  unit_price: number;
  msrp?: number | null;
  min_order_qty: number;
  lead_time: string;
  status: string;
};

export const EMPTY_SKU: SKUItem = {
  sku_number: "",
  description: "",
  unit_of_measure: "EA",
  unit_price: 0,
  msrp: null,
  min_order_qty: 1,
  lead_time: "14 days",
  status: "Active",
};

export const SKU_STATUSES = ["Active", "Inactive", "Pending", "Discontinued"] as const;
export const SKU_UOMS = ["EA", "CS", "BX", "PK", "PR", "KG", "LB"] as const;
